### DISCUSSION TITLE
How to obtain the access_token of the identify provider authorization server？

### DISCUSSION THREAD

**Discussion by wyhdil** [keycloak/keycloak#23961] [Q&A] [answered]
Hello, I have now configured self-hosted gitlab as the identify provider.

![f1449000c7c13148d9292fb96d6c8562](https://github.com/keycloak/keycloak/assets/69418800/d7e99cad-2f2b-4f82-b8af-d211c64e60d0)
When the user logs in through gitlab authorization, in theory an access_token should be returned to keycloak，and my question is, how do I get the access_token returned by gitlab to keycloak?

Because I want to get the information about the user's settings on gitlab, I want to get the access_token issued by gitlab.
Or is there any other way to obtain the information set by the user on gitlab?

Looking forward to your answer, thank you very much！

**Comment by danielFesenmeyer** [comment, marked as answer]
Afaik, this functionality is provided by the API - see https://www.keycloak.org/docs/latest/server_admin/#retrieving-external-idp-tokens.
